doc/source/examining/loading_data.rst

```diff
@@ -2030,6 +2038,35 @@ Note: for backward compatibility, particles from the
 default (including dark matter, stars, tracer particles, …). Sink
 particles have the particle type ``sink``.
 
+Particle automatic filtering
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If your RAMSES version is more recent than `stable_17_09`, it is
+possible to tell yt to filter the particles in your dataset. This is
+not done by default is it requires to read all the particles and may
+be small. To use this feature, run
+
+.. code-block:: python
+
+   ds = yt.load('ramses_new_format/output_00011/info_00011.txt')
+
+   # This will load the particle types automatically
+   ds.add_ptypes()
+
+
+Adding custom particle fields
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It is possible to add support for particle fields. For this, one
+should tweak
+:func:`~yt.frontends.ramses.io._read_part_file_descriptor` to include
+the field as well as its data type to the assoc list, following the
+convention from
+`python struct module <https://docs.python.org/3.5/library/struct.html#format-characters>`_.
+For example, to add support for a longint field named
+`my_custom_field`, one would add `('my_custom_field', 'l')` to `assoc`.
```

Comment:
Is this something people would do at runtime? Maybe an explicit example would be helpful.

Reply:
If for some reason the user is handling an output where there are some particle fields that can't be automatically detected, it gives the user the possibility to force their detection. I'll add an example.

Reply:
Sorry I misunderstood the question: this is not done at runtime, it is just a suggestion for advanced user to have yt detecting their field automatically as well as reading it in the right format.